core/node/da_dispatcher/src/da_dispatcher.rs

```diff
@@ -0,0 +1,147 @@
+        let pool = self.pool.clone();
+        loop {
+            let mut conn = pool.connection_tagged("da_dispatcher").await.unwrap();
+
+            if *stop_receiver.borrow() {
+                tracing::info!("Stop signal received, da_dispatcher is shutting down");
+                break;
+            }
+
+            if let Err(err) = self.dispatch(&mut conn).await {
+                tracing::warn!("dispatch error {err:?}");
+            }
+            if let Err(err) = self.poll_for_inclusion(&mut conn).await {
+                tracing::warn!("poll_for_inclusion error {err:?}");
+            }
+
+            drop(conn);
+            tokio::time::sleep(self.config.polling_interval()).await;
+        }
+        Ok(())
+    }
+
+    async fn dispatch(&self, conn: &mut Connection<'_, Core>) -> anyhow::Result<()> {
+        let batches = conn
+            .blocks_dal()
+            .get_ready_for_da_dispatch_l1_batches(self.config.query_rows_limit() as usize)
+            .await?;
+
+        for batch in batches {
+            let dispatch_latency = METRICS.blob_dispatch_latency.start();
+            let dispatch_response = retry(self.config.max_retries(), || {
+                self.client
+                    .dispatch_blob(batch.l1_batch_number.0, batch.pubdata.clone())
+            })
+            .await
+            .map_err(Error::msg)?;
+            dispatch_latency.observe();
+
+            conn.blocks_dal()
+                .insert_l1_batch_da(batch.l1_batch_number, dispatch_response.blob_id)
+                .await?;
+
+            METRICS
+                .last_known_l1_batch
+                .set(batch.l1_batch_number.0 as usize);
+            METRICS.blob_size.observe(batch.pubdata.len());
+            tracing::info!(
+                "Dispatched a DA for batch_number: {}",
+                batch.l1_batch_number
+            );
+        }
+
+        Ok(())
+    }
+
+    async fn poll_for_inclusion(&self, conn: &mut Connection<'_, Core>) -> anyhow::Result<()> {
+        let storage_da = conn.blocks_dal().get_da_blob_awaiting_inclusion().await?;
+
+        if let Some(storage_da) = storage_da {
+            let inclusion_data = self
+                .client
+                .get_inclusion_data(storage_da.blob_id.clone().unwrap())
+                .await
+                .map_err(Error::msg)?;
+
+            if let Some(inclusion_data) = inclusion_data {
+                conn.blocks_dal()
+                    .save_l1_batch_inclusion_data(
+                        L1BatchNumber(storage_da.l1_batch_number as u32),
+                        inclusion_data.data,
+                    )
+                    .await?;
+
+                METRICS.inclusion_latency.observe(Duration::from_secs(
+                    (Utc::now().timestamp() - storage_da.created_at.timestamp()) as u64,
+                ));
+
+                tracing::info!(
+                    "Received an inclusion data for a batch_number: {}",
```

Comment:
nit: can add more data here if it makes sense

Reply:
added